codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionDecoder.java

```diff
@@ -510,31 +499,15 @@ public void onUnknownFrame(ChannelHandlerContext ctx, byte frameType, int stream
             onUnknownFrame0(ctx, frameType, streamId, flags, payload);
         }
 
-        /**
-         * Indicates whether or not frames for the given stream should be ignored based on the state of the
-         * stream/connection.
-         */
-        private boolean shouldIgnoreFrame(Http2Stream stream, boolean allowResetSent) {
-            if (connection.goAwaySent() &&
-                    (stream == null || connection.remote().lastStreamCreated() <= stream.id())) {
-                // Frames from streams created after we sent a go-away should be ignored.
-                // Frames for the connection stream ID (i.e. 0) will always be allowed.
-                return true;
-            }
-
-            // Also ignore inbound frames after we sent a RST_STREAM frame.
-            return stream != null && !allowResetSent && stream.isResetSent();
+        private boolean streamCreatedAfterGoAwaySent(Http2Stream stream) {
```

Comment:
I think you should break this into 2 separate checks:

`streamMayHaveExistedPreviously()` and `streamCreatedAfterGoAwaySent()`.  

It's a little confusing since I wouldn't expect `streamCreatedAfterGoAwaySent()` to be true for a historical stream.